components/script/dom/gpubuffer.rs

```diff
@@ -75,8 +83,11 @@ impl GPUBuffer {
             device,
             buffer,
             mapping,
+            mapped_ranges: DomRefCell::new(None),
```

Comment:
do all of these need to be separate? Can we do `DomRefCell<Option<SomeStructure>>` with all the fields in the `SomeStructure`?